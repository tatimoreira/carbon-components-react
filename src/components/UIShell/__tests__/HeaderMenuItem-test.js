import React from 'react';
import { mount } from 'enzyme';
import { HeaderMenuItem } from '../';

describe('HeaderMenuItem', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      className: 'custom-class',
      children: 'Menu Item',
      role: 'none',
      ref: jest.fn(),
    };
  });

  it('should render', () => {
    const wrapper = mount(<HeaderMenuItem {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(mockProps.ref).toHaveBeenCalledTimes(1);
  });
});
