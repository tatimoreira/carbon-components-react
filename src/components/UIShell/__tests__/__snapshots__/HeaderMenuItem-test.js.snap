// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuItem should render 1`] = `
<ForwardRef
  className="custom-class"
  role="none"
>
  <HeaderMenuItem
    className="custom-class"
    innerRef={null}
    role="none"
  >
    <li
      className="custom-class"
      role="none"
    >
      <a
        className="bx--platform-header__menu-item"
        role="menuitem"
      >
        <span
          className="bx--text-truncate--end"
        >
          Menu Item
        </span>
      </a>
    </li>
  </HeaderMenuItem>
</ForwardRef>
`;
